xcode-compat/src/main/kotlin/org/jetbrains/kotlin/xcodecompat/FrameworkXcodeProjectTask.kt

```diff
@@ -0,0 +1,328 @@
+
+open class FrameworkXcodeProjectTask : DefaultTask() {
+
+    @Input
+    lateinit var frameworkName: String
+
+    @Input
+    lateinit var gradleTaskName: String
+
+    val projectGroup
+        @Input get() = project.group.toString()
+
+    @OutputDirectory
+    val outputDirectory: File = project.projectDir.resolve("xcodeproj")
+
+    @TaskAction
+    fun generate() {
+        val gradleWrapper = (project.rootProject.tasks.getByName("wrapper") as? Wrapper)?.scriptFile
+        require(gradleWrapper != null && gradleWrapper.exists()) {
+            """
+            The Gradle wrapper is required to run the build from Xcode.
+
+            Please run the same command with `-P$GENERATE_WRAPPER_PROPERTY=true` or run the `:wrapper` task to generate the wrapper manually.
+
+            See details about the wrapper at https://docs.gradle.org/current/userguide/gradle_wrapper.html
+            """.trimIndent()
+        }
+
+        outputDirectory.mkdirs()
+        val infoPlistRelative = "Info.plist"
+        outputDirectory.resolve(infoPlistRelative).writeText("""
+            <?xml version="1.0" encoding="UTF-8"?>
+            <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+            <plist version="1.0">
+            <dict>
+                <key>CFBundleDevelopmentRegion</key>
+                <string>${'$'}(DEVELOPMENT_LANGUAGE)</string>
+                <key>CFBundleExecutable</key>
+                <string>${'$'}(EXECUTABLE_NAME)</string>
+                <key>CFBundleIdentifier</key>
+                <string>${'$'}(PRODUCT_BUNDLE_IDENTIFIER)</string>
+                <key>CFBundleInfoDictionaryVersion</key>
+                <string>6.0</string>
+                <key>CFBundleName</key>
+                <string>${'$'}(PRODUCT_NAME)</string>
+                <key>CFBundlePackageType</key>
+                <string>FMWK</string>
+                <key>CFBundleShortVersionString</key>
+                <string>1.0</string>
+                <key>CFBundleVersion</key>
+                <string>${'$'}(CURRENT_PROJECT_VERSION)</string>
+            </dict>
+            </plist>
+        """.trimIndent())
+
+        val xcodeprojDirectory = outputDirectory.resolve(frameworkName + ".xcodeproj")
+        val bundleIdentifier: String = projectGroup.let { if (it.isEmpty()) frameworkName else "$it.$frameworkName" }
+
+        xcodeprojDirectory.mkdirs()
+        xcodeprojDirectory.resolve("project.pbxproj").writeText("""
+            // !${'$'}*UTF8*${'$'}!
+            {
+                archiveVersion = 1;
+                classes = {
+                };
+                objectVersion = 46;
+                objects = {
+
+            /* Begin PBXFileReference section */
+                    62E985139ABA658F4C6C577F /* $frameworkName.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = $frameworkName.framework; sourceTree = BUILT_PRODUCTS_DIR; };
```

Comment:
Is it safe to hardcode `62E985139ABA658F4C6C577F` strings like that? Could it, in theory, affect any existing Xcode project?

Reply:
That's what I'm wondering about too. I hope @stigger could help us with that.

Reply:
These identifiers must be unique in the scope of a project, and since this is a new project, it should be fine.

Reply:
How do you feel about a scenario, where one wants to mix several generated projects (via the task) in a bigger project?

Reply:
Not sure if this project can be a part of bigger project in that sense. @stigger?

Reply:
to clarify - one may create something with the task and then move that into a bigger project (via Xcode tools if possible or just manually). +1, I'm not sure it is possible or people do so

Reply:
I believe it should work fine, but better test it to be sure.

Reply:
Made ids more unique instead.